File: ca_reqres/test_mysql.go
//go:build ignore

package main

import (
	"database/sql"
	"fmt"
	"time"

	_ "github.com/go-sql-driver/mysql"
)

func main() {
	dsn := "emqIiIjEB3s7HKTE@tcp(mysql6.sqlpub.com:3311)/oneapi_aganhui"

	fmt.Println("正在连接 MySQL...")
	fmt.Printf("DSN: %s

", dsn)

	db, err := sql.Open("mysql", dsn)
	if err != nil {
		fmt.Printf("[FAIL] 打开连接失败: %v
", err)
		return
	}
	defer db.Close()

	db.SetConnMaxLifetime(10 * time.Second)
	db.SetMaxOpenConns(1)
	db.SetMaxIdleConns(1)

	start := time.Now()
	if err := db.Ping(); err != nil {
		fmt.Printf("[FAIL] Ping 失败: %v
", err)
		return
	}
	elapsed := time.Since(start)

	fmt.Printf("[OK] 连接成功！耗时: %v
", elapsed)

	// 查询服务器版本
	var version string
	if err := db.QueryRow("SELECT VERSION()").Scan(&version); err != nil {
		fmt.Printf("[WARN] 查询版本失败: %v
", err)
	} else {
		fmt.Printf("[OK] MySQL 版本: %s
", version)
	}

	// 查询当前数据库
	var dbName string
	if err := db.QueryRow("SELECT DATABASE()").Scan(&dbName); err != nil {
		fmt.Printf("[WARN] 查询数据库名失败: %v
", err)
	} else {
		fmt.Printf("[OK] 当前数据库: %s
", dbName)
	}
}

File: ca_reqres/test_mysql.py
#!/usr/bin/env python3
# MySQL 连接测试脚本

import time
import sys

try:
    import pymysql
except ImportError:
    print("[FAIL] 未找到 pymysql，请先运行: pip3 install pymysql")
    sys.exit(1)

# DSN: emqIiIjEB3s7HKTE@tcp(mysql6.sqlpub.com:3311)/oneapi_aganhui
HOST     = "mysql6.sqlpub.com"
PORT     = 3311
USER     = "aganhui"
PASSWORD = "emqIiIjEB3s7HKTE"
DATABASE = "oneapi_aganhui"

print("正在连接 MySQL...")
print(f"  Host : {HOST}:{PORT}")
print(f"  User : {USER}")
print(f"  DB   : {DATABASE}")
print()

try:
    start = time.time()
    conn = pymysql.connect(
        host=HOST,
        port=PORT,
        user=USER,
        password=PASSWORD,
        database=DATABASE,
        connect_timeout=10,
    )
    elapsed = (time.time() - start) * 1000
    print(f"[OK] 连接成功！耗时: {elapsed:.1f} ms")

    with conn.cursor() as cur:
        cur.execute("SELECT VERSION()")
        version = cur.fetchone()[0]
        print(f"[OK] MySQL 版本: {version}")

        cur.execute("SELECT DATABASE()")
        db_name = cur.fetchone()[0]
        print(f"[OK] 当前数据库: {db_name}")

        cur.execute("SHOW TABLES")
        tables = [row[0] for row in cur.fetchall()]
        print(f"[OK] 表数量: {len(tables)}")
        if tables:
            print(f"     表列表: {', '.join(tables)}")

    conn.close()

except pymysql.err.OperationalError as e:
    print(f"[FAIL] 连接失败: {e}")
    sys.exit(1)
except Exception as e:
    print(f"[FAIL] 未知错误: {e}")
    sys.exit(1)

